src/en/news/blog/2022/bucket-notifications-with-knative-and-rook-on-minikube2/index.md

```diff
@@ -0,0 +1,330 @@
+cat << EOF | kubectl apply -f -
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: knative-serving
+---
+apiVersion: operator.knative.dev/v1alpha1
+kind: KnativeServing
+metadata:
+  name: knative-serving
+  namespace: knative-serving
+EOF
+```
+
+The networking layer (using Kourier);
+
+```bash
+cat << EOF | kubectl apply -f -
+apiVersion: operator.knative.dev/v1alpha1
+kind: KnativeServing
+metadata:
+  name: knative-serving
+  namespace: knative-serving
+spec:
+  ingress:
+    kourier:
+      enabled: true
+  config:
+    network:
+      ingress-class: "kourier.ingress.networking.knative.dev"
+EOF
+```
+
+The "Ceph Source" is now packaged as part of Knative (yay!), so there is no need to install it from [source](https://github.com/knative-sandbox/eventing-ceph). We will install the "eventing" component with the "Ceph Source":
+
+```bash
+cat << EOF | kubectl apply -f -
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: knative-eventing
+---
+apiVersion: operator.knative.dev/v1alpha1
+kind: KnativeEventing
+metadata:
+  name: knative-eventing
+  namespace: knative-eventing
+spec:
+  source:
+    ceph:
+      enabled: true
+EOF
+```
+
+### Rook
+
+Deploy the Rook operator (can also look at the [quickstart guide](https://rook.io/docs/rook/v1.8/quickstart.html)):
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/rook/rook/release-1.8/deploy/examples/crds.yaml
+kubectl apply -f https://raw.githubusercontent.com/rook/rook/release-1.8/deploy/examples/common.yaml
+kubectl apply -f https://raw.githubusercontent.com/rook/rook/release-1.8/deploy/examples/operator.yaml
+```
+
+Now the "test" (single node) Ceph cluster:
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/rook/rook/release-1.8/deploy/examples/cluster-test.yaml
+```
+
+And make sure that the OSDs and MONs are up and running:
+
+```bash
+kubectl -n rook-ceph get pod
+```
+
+And last, the "test" (single node) object store:
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/rook/rook/release-1.8/deploy/examples/object-test.yaml
+```
+
+## Really Easy as a YAML
+
+Lets setup **everything** from YAMLs:
+
+### Ceph Source in Knative
+
+Create the Ceph Source CR and service (that would be used as the endpoint for the bucket notification topic):
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/knative-sandbox/eventing-ceph/release-1.2/samples/ceph-source.yaml
+```
+
+```bash
+cat << EOF | kubectl apply -f -
+apiVersion: v1
+kind: Service
+metadata:
+  name: my-ceph-source-svc
+spec:
+  selector:
+    eventing.knative.dev/sourceName: my-ceph-source
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 8888
+EOF
+```
+
+The receiver (sink) for the event would be a generic event display pod:
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/knative-sandbox/eventing-ceph/release-1.2/samples/event-display.yaml
+```
+
+Note that the "event-display" pod would start, but terminate once it sees that it has no events to handle.
```

Comment:
Note that the "event-display" pod will start but terminates with no events to handle.